File: system/multitier/backend-java/src/integrationTest/java/com/mycompany/myshop/backend/AbstractIntegrationTest.java
package com.mycompany.myshop.backend;

import org.springframework.boot.testcontainers.service.connection.ServiceConnection;
import org.testcontainers.containers.PostgreSQLContainer;
import org.testcontainers.junit.jupiter.Container;
import org.testcontainers.junit.jupiter.Testcontainers;

@Testcontainers
public abstract class AbstractIntegrationTest {

    @Container
    @ServiceConnection
    static final PostgreSQLContainer<?> POSTGRES =
        new PostgreSQLContainer<>("postgres:16-alpine")
            .withDatabaseName("app")
            .withUsername("app")
            .withPassword("app");
}

File: system/monolith/dotnet/Tests/Integration/AbstractIntegrationTest.cs
using Microsoft.EntityFrameworkCore;
using MyCompany.MyShop.Monolith.Data;
using Testcontainers.PostgreSql;
using Xunit;

namespace MyCompany.MyShop.Monolith.Tests.Integration;

public abstract class AbstractIntegrationTest : IAsyncLifetime
{
    private readonly PostgreSqlContainer _postgres = new PostgreSqlBuilder()
        .WithImage("postgres:16-alpine")
        .WithDatabase("app")
        .WithUsername("app")
        .WithPassword("app")
        .Build();

    protected AppDbContext DbContext { get; private set; } = null!;

    public async Task InitializeAsync()
    {
        await _postgres.StartAsync();

        var options = new DbContextOptionsBuilder<AppDbContext>()
            .UseNpgsql(_postgres.GetConnectionString())
            .Options;

        DbContext = new AppDbContext(options);
        await DbContext.Database.EnsureCreatedAsync();
    }

    public async Task DisposeAsync()
    {
        await DbContext.DisposeAsync();
        await _postgres.DisposeAsync();
    }
}
